### DISCUSSION TITLE
Supports serial

### DISCUSSION THREAD

**Discussion by itnewgay** [1Remote/1Remote#478] [Ideas]
有时候要配置网络设备。希望能支持serial 串口功能。

**Comment by VShawn** [comment]
check the latest 1Remote for serial protocol, I'm not sure it can works since I dont have any test environment for serial.

**Reply by itagagaki** [reply]
It works fine from what I've been able to test on my end :)

It would be even better if more options were added, such as font family and size, cursor style, etc..
PS:
Oh, I see that the serial terminal also uses KiTTY. Then, as with #785, the question is how to change the KiTTY settings.

**Reply by itagagaki** [reply]
I was able to change some configurations of the serial terminal by specifying the following file to the KiTTY Session for Serial, just as I wrote in my comment to #785.

```
Colour0\0,187,63\
Colour5\255,63,0\
CurType\1\
BlinkCur\1\
Font\Arial\
FontHeight\10\
```

However, `Colour0`, `Colour5` and `FontHeight` did not change. This is probably because 1Remote's protocol runner setting is further overridden. there is no runner setting for Serial, but the change to the SSH runner setting was also reflected in Serial. (I will file some Issues on this).

**Reply by VShawn** [reply]
> However, Colour0, Colour5 and FontHeight did not change. This is probably because 1Remote's protocol runner setting is further overridden.

here

https://github.com/1Remote/1Remote/blob/main/Ui/Utils/KiTTY/IKittyConnectable.cs#L46-L102

The session file is loaded first, and then the setting parameters from 1Remote are added to the list. That’s why `Colour0, Colour5 and FontHeight did not change`.

**Reply by itagagaki** [reply]
I think we should change the order, what do you think?

**Reply by VShawn** [reply]
Well. I dont know, It's a little weird that the theme will be overwritten, but I don't think it's a big deal.

BTW, you can't switch the order completely, for example, the hostname in the GUI should not be overwritten by the session file.

**Reply by itagagaki** [reply]
Current priorities are:

user's session file (all configurable) < Configuration items provided by 1Remote GUI (e.g. connection info) and some implicit configuration items < Theme

It depends on the mindset, but this makes sense too.
Okay. I withdraw the proposal to change this.

**Reply by VShawn** [reply]
Actually, I don't have any particular insistence on this, except that connection info cannot be overwritten, others is ok either way.

Theme < session file < connection info, is ok to me.

BTW, A same dilemma in RDP：

![image](https://github.com/user-attachments/assets/44769cc1-dd60-4163-bbe1-4f895a3601b1)
